File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
           int max=Integer.MIN_VALUE;
         

        if(prices.length<=1)return 0;
         int buy=prices[0];
        for (int i = 0; i < prices.length; i++) {
            buy =Math.min(buy,prices[i]);
            //for (int j = i; j < prices.length; j++) {
               max= Math.max(prices[i]-buy,max);
                
            //}//
             
        }
        return max;
    }
}

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        r = 0
        
        buy=prices[0]
        for i in range(len(prices)):
            buy=min(prices[i],buy)
            r = max(r, prices[i] - buy)
        return r